### ISSUE TITLE
FileLength Add ignorePattern Property

### ISSUE THREAD

**Issue by atb-brown** [checkstyle/checkstyle#11558]
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac Test.java
# Compiles ✅

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <module name="FileLength">
        <property name="max" value="10"/>
        <property name="fileExtensions" value="java"/>
        <!-- This would be a new property. -->
        <!-- I may not have these regex patterns perfect, but the intent would
              be to ignore Java comments and JavaDocs -->
        <property name="ignorePattern" value="(^[ ]+//.*$)|(^[ ]/\*\*.*\*/$)"/>
    </module>

</module>


/var/tmp $ cat Test.java
/**
 * This class would pass with the ignorePattern property, but it
 * will fail without it.
 */
public class Test {

  /**
   * A simple hello world method.
   **/
   void helloWorld() {
    // Say hi!
    System.out.println("Hello world.");
  }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml Test.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module FileLength - Property 'ignorePattern' does not exist, please check the documentation
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property 'ignorePattern' does not exist, please check the documentation
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:227)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:194)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.

```

**Is your feature request related to a problem? Please describe.**
No problem with existing functionality.

**Describe the solution you'd like**
It would be cool if there was a [FileLength property](https://checkstyle.sourceforge.io/config_sizes.html#FileLength_Properties) similar to the ignorePattern [LineLength property](https://checkstyle.sourceforge.io/config_sizes.html#LineLength_Properties).

When the `ignorePattern` property is included on the `FileLength` check, it would ignore all lines that match the pattern when counting the lines in a file.

**Additional context**
For me (and as suggested in the example) this would be really helpful because it would allow me to ignore Javadocs and comments in my Java classes. I think this would incentivize others on my team to be thorough with their Javadocs and not punish them for writing long Javadocs.

**Comment by Vyom-Yadav** [comment]
Please remove lines from `Still not clear ???`, other than that looks good to me.

**Comment by Vyom-Yadav** [comment]
@romani @rnveach @nick-mancuso @strkkk @pbludov Please share your thoughts.

**Comment by pbludov** [comment]
@atb-brown Why don't you use [NCSS check](https://checkstyle.sourceforge.io/config_metrics.html#JavaNCSS)? It counts file lines **excluding** blank lines and comments.

**Comment by atb-brown** [comment]
> @pbludov 
> use [NCSS check](https://checkstyle.sourceforge.io/config_metrics.html#JavaNCSS)?

Ahh, interesting. I was unaware of that rule. Using that in combination with the `fileLength` rule seems like it would do the trick for my needs with Java. I also, however, would like to do something similar with non-Java files (in my case, XSLT).

I figure it would be easier to have support for custom regular expressions than it would be to have support for each an every language. But I could just be guilty of wishful thinking, and this NCSS rule is certainly better than nothing.

**Comment by romani** [comment]
If you are so comment focused in code, you have to extend limit for file as you might presume that comment would take 25% of your code.

File length validation is always very rough measurement of code quality or level before split. It helps to guide people but it never absolute must. Even checkstyle project suppress violation from this Check in very certain cases.
